website/docs/plugin/framework/resources/default.mdx

```diff
@@ -0,0 +1,99 @@
+| [`schema.ObjectAttribute`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema#ObjectAttribute) / | [`schema.SingleNestedAttribute`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema#SingleNestedAttribute) |  [`resource/schema/objectdefault` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema/objectdefault) |
+| [`schema.SetAttribute`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema#SetAttribute) / | [`schema.SetNestedAttribute`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema#SetNestedAttribute) |  [`resource/schema/setdefault` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema/setdefault) |
+| [`schema.StringAttribute`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema#StringAttribute) |  [`resource/schema/stringdefault` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault) |
+
+### Custom Default Implementations
+
+To create an attribute default, you must implement the one of the [`resource/schema/defaults` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/resource/schema/defaults) interfaces. For example:
+
+```go
+// timeDefaultValue is a default that sets the value for a types.StringType
+// attribute to the current time when it is not configured. The attribute
+// must be marked as Optional and Computed. When setting the state during
+// the resource Create, Read, or Update methods, this value must also be
+// included or the Terraform CLI will generate an error.
+type timeDefaultValue struct {
+	time time.Time
+}
+
+// Description returns a plain text description of the default's behavior, suitable for a practitioner to understand its impact.
+func (d timeDefaultValue) Description(ctx context.Context) string {
+	return fmt.Sprintf("If value is not configured, defaults to a string representation of the current time")
+}
+
+// MarkdownDescription returns a markdown formatted description of the default's behavior, suitable for a practitioner to understand its impact.
+func (d timeDefaultValue) MarkdownDescription(ctx context.Context) string {
+	return fmt.Sprintf("If value is not configured, defaults to a string representation of the current time")
+}
+
+// DefaultString runs the logic of the default. Access to the path is available in `req`, while
+// `resp` contains fields for updating the planned value, and returning diagnostics.
+func (d timeDefaultValue) DefaultString(_ context.Context, req defaults.StringRequest, resp *defaults.StringResponse) {
+resp.PlanValue = types.StringValue(d.time.String())
```

Comment:
Nit: [`(time.Time).String()`](https://pkg.go.dev/time#Time.String) returns a specific formatted string:

>The returned string is meant for debugging; for a stable serialized representation, use t.MarshalText, t.MarshalBinary, or t.Format with an explicit format string.

While `Format(time.RFC3339)` can ensure it follows the Terraform conventional RFC3339 timestamp format.

```suggestion
	resp.PlanValue = types.StringValue(d.time.Format(time.RFC3339))
```